clients/unshaded/src/main/java/tachyon/client/block/BlockWorkerClientPool.java

```diff
@@ -60,6 +66,12 @@ public void close() {
 
   @Override
   public void release(WorkerClient workerClient) {
+    try {
+      // Heartbeat to send the client metrics.
+      workerClient.sessionHeartbeat();
+    } catch (IOException ioe) {
+      LOG.warn("Failed sending client metrics before releasing the worker client", ioe);
+    }
```

Comment:
No, we don't need to clear the metrics here.  All the worker clients share a single client metrics object stored in the client context. The metrics will be cleared when doing the heartbeat. Even if an exception happens, the metrics that are not cleared will be sent by the next heartbeat.